gammapy/maps/wcsnd.py

```diff
@@ -589,6 +585,63 @@ def get_spectrum(self, region=None, func=np.nansum, weights=None):
 
         return self.to_region_nd_map(region=region, func=func, weights=weights)
 
+    def binary_erosion(self, margin, from_edges=False):
+        """Binary erosion of boolean mask removing a given margin
+        Parameters
+        ----------
+        margin : tuple of `~astropy.units.Quantity`
+            Angular sizes of the margin in (lon, lat) in that specific order.
+            If only one value is passed, the same margin is applied in (lon, lat).
+        from_edge : bool
+            If True apply the margin only from edges (Default is False).
+        """
+        if self.data.dtype != bool:
+            raise (TypeError, "Binary erosion should be applied only on boolean mask")
+        if from_edges:
+            mask_margin = np.ones(self.data.shape[-2:], dtype=bool)
+        else:
+            # TODO: should we apply the erosin independently to each other dim
```

Comment:
Again I would make it more similar to `.convolve()` and `.smooth()`: so for now apply the erosion, per image plane. This offers the possibility to handle energy dependent "kernels" in future (which seems reasonable, as e.g. the PSF size is energy dependent as well.).